server/protocols/webadmin/webadmin-mail-over-web/src/main/java/org/apache/james/webadmin/request/BodyPartProps.java

```diff
@@ -0,0 +1,89 @@
+package org.apache.james.webadmin.request;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class BodyPartProps {
+
+    @JsonCreator
+    public BodyPartProps(
+            @JsonProperty("type") String type,
+            @JsonProperty("filename") String filename,
+            @JsonProperty("content") String content,
+            @JsonProperty("encoding") String encoding,
+            @JsonProperty("disposition") String disposition,
+            @JsonProperty("headers") Map<String, String> headers) {
+
+        Optional.ofNullable(type).ifPresent(this::setType);
+        this.filename = filename;
+        Optional.ofNullable(content).ifPresent(this::setContent);
+        this.encoding = encoding;
+        this.disposition = disposition;
+        Optional.ofNullable(headers).ifPresent(this::setHeaders);
+    }
+
+    private String type = "text/plain";
+
+    private String filename;
+
+    private String content;
+
+    private String encoding;
+
+    private String disposition;
+
+    private Map<String, String> headers = new HashMap<>();
```

Comment:
We tend to use immutable POJOs, using final fields (no setters) and immutable datastructures like ImmutableMap (guava)